src/compiler/parser.ts

```diff
@@ -2860,10 +2860,24 @@ namespace ts {
             return type;
         }
 
+        function isStartOfTypeofClassExpression() {
+            return token() === SyntaxKind.ClassKeyword ||
+                token() === SyntaxKind.AbstractKeyword && lookAhead(() => nextToken() === SyntaxKind.ClassKeyword && !scanner.hasPrecedingLineBreak());

```

Comment:
As I mention in my comment in `checker.ts`, is there a reason we don't permit `abstract` for class expressions? It seems like we should always do this, not just for `typeof`.